lib/src/results.dart

```diff
@@ -135,6 +135,85 @@ extension RealmResultsOfObject<T extends RealmObjectBase> on RealmResults<T> {
   }
 }
 
+class _SubscribedRealmResult<T extends RealmObject> extends RealmResults<T> {
+  final String? subscriptionName;
+
+  _SubscribedRealmResult._(RealmResults<RealmObject> results, {this.subscriptionName})
+      : super._(
+          results.handle,
+          results.realm,
+          results.metadata,
+        );
+}
+
+extension RealmResultsOfRealmObject<T extends RealmObject> on RealmResults<T> {
+  /// Adds this [RealmResults] query to the set of active subscriptions.
+  /// The query will be joined via an OR statement with any existing queries for the same type.
+  ///
+  /// If a [name] is given this allows you to later refer to the subscription by name,
+  /// e.g. when calling [MutableSubscriptionSet.removeByName].
+  ///
+  /// If [update] is specified to `true`, then any existing query
+  /// with the same name will be replaced.
+  /// Otherwise a [RealmException] is thrown, in case of duplicates.
+  ///
+  /// [WaitForSyncMode] specifies how to wait or not wait for subscribed objects to be downloaded.
+  /// The default value is [WaitForSyncMode.onCreation].
+  ///
+  /// The [timeout] is the maximum duration to wait for objects to be downloaded.
+  /// If the time exceeds this limit, the [RealmResults] is returned and the download
+  /// continues in the background.
+  ///
+  /// {@category Sync}
+  Future<RealmResults<T>> subscribe({
+    String? name,
+    WaitForSyncMode waitForSyncMode = WaitForSyncMode.onCreation,
+    Duration? timeout,
```

Comment:
I would prefer to take a `CancellationToken`.. people can pass one with a timeout.. and maybe even default to using one with a timeout, if none is passed, but we should discuss that, as it is not inline with the other SDKs as I understand.

Reply:
Done